### ISSUE TITLE
Attempting to solve player sync issues

### ISSUE THREAD

**Issue by valkyrienyanko** [Valks-Games/sankari#108]
```
Unhandled Exception:
System.Collections.Generic.KeyNotFoundException: The given key '0' was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x0001e] in <c84d94161bca414384015e66544862cf>:0
  at Sankari.SPacketPlayerPositions.Handle () [0x0007d] in <d935e8a3f81847c8927c287d3a028360>:0
  at Sankari.GodotCommands.Update () [0x00142] in <d935e8a3f81847c8927c287d3a028360>:0
  at Sankari.Netcode.Net.Update () [0x0006e] in <d935e8a3f81847c8927c287d3a028360>:0
  at Sankari.GameManager.Update () [0x00068] in <d935e8a3f81847c8927c287d3a028360>:0
  at Sankari.Linker._Process (System.Single delta) [0x00074] in <d935e8a3f81847c8927c287d3a028360>:0
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in <c84d94161bca414384015e66544862cf>:0
  at Godot.GodotSynchronizationContext.ExecutePendingContinuations () [0x00021] in <8c99985ff12947baa88b379583357ac3>:0
  at Godot.GodotTaskScheduler.Activate () [0x0000e] in <8c99985ff12947baa88b379583357ac3>:0
  at (wrapper native-to-managed) Godot.GodotTaskScheduler.Activate(Godot.GodotTaskScheduler,System.Exception&)
```

**Comment by valkyrienyanko** [comment]
![image](https://user-images.githubusercontent.com/6277739/189817641-28d83d45-256e-4c88-9d37-404c7c4d37b9.png)

**Comment by valkyrienyanko** [comment]
![image](https://user-images.githubusercontent.com/6277739/189817733-ea773354-f274-4a94-a5b5-b2634d923ab2.png)

**Comment by valkyrienyanko** [comment]
![image](https://user-images.githubusercontent.com/6277739/189818832-41cf53ba-81e4-4a7b-83b6-2fc1afef2a8e.png)

**Comment by valkyrienyanko** [comment]
![image](https://user-images.githubusercontent.com/6277739/189824595-22bd4da8-854a-40fd-967e-3e5b73879a32.png)

**Comment by valkyrienyanko** [comment]
This does not look right to me. 

This implies that OtherClient receives OtherClient and only OtherClient position

ImHost should receive OtherClient position

**Comment by valkyrienyanko** [comment]
```cs
// ImHost needs to only see OtherClient position
// OtherClient needs to only see ImHost position

// Lets say 'player' = ImHost
foreach (var player in Players)
{
    // [ImHost, OtherClient]
    var playerPositions = new Dictionary<byte, DataPlayer>(Players).ToDictionary(x => x.Key, x => x.Value.Position);
    
    // Remove ImHost from the list of player positions
    // [OtherClient]
    playerPositions.Remove(player.Key);

    // Send OtherClient position to everyone but ImHost
    // Sending OtherClient position to OtherClient (THIS IS WHERE THE ISSUE IS)
    SendToOtherPlayers((uint)player.Key, ServerPacketOpcode.PlayerPositions, new SPacketPlayerPositions 
    {
        PlayerPositions = playerPositions
    });
}
```

**Comment by valkyrienyanko** [comment]
Actually nvm this is correct lol. What am I on about. The issue must be somewhere else.

"Sending OtherClient position to OtherClient". This is normal lol. This is what I want ;-;

**Comment by valkyrienyanko** [comment]
![image](https://user-images.githubusercontent.com/6277739/189828594-ece7caa2-b5a8-4407-8340-e759832ad03a.png)

**Comment by valkyrienyanko** [comment]
```cs
// Sending OtherClient position to OtherClient
``` THIS IS WRONG. It should be sending every other position except for OtherClient position to OtherClient

**Comment by valkyrienyanko** [comment]
![image](https://user-images.githubusercontent.com/6277739/189831996-04668d51-326e-4c05-b419-c8c7f557fd8a.png)

So the error is definitely here!

And don't forget this also needs to be optimized!

![image](https://user-images.githubusercontent.com/6277739/189832129-5c4b2447-63ae-4e5e-b72e-58e9927fc3c7.png)

**Comment by valkyrienyanko** [comment]
Solved 7ac59ab19cadb27f3a09c5910b858d8fbd8a2732
